File: backend/src/main/kotlin.nikolay/otdo/services/ServicesService.kt
package otdo.services

import otdo.services.model.Service

@org.springframework.stereotype.Service
class ServicesService(private val servicesRepository: ServicesRepository) {
    fun getServices(): List<Service> {
        return servicesRepository.getServices()
    }

    fun saveService(service: Service): Long {
        return if (service.id == null) {
            servicesRepository.createNewService(service)
        } else {
            servicesRepository.editService(service)
            service.id
        }
    }

    fun deleteService(id: Long) {
        servicesRepository.deleteService(id)
    }
}

File: frontend/src/app/pages/services/services.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Service} from "./model/Service";
import {environment} from "../../../environments/environment";

@Injectable({
  providedIn: 'root'
})
export class ServicesService {
  private readonly domain: string;

  constructor(private http: HttpClient) {
    this.domain = environment._SERVER + '/services'
  }

  getServices(): Observable<Service[]> {
    return this.http.get<Service[]>(this.domain)
  }

  saveService(service: Service): Observable<number> {
    return this.http.post<number>(this.domain, service);
  }
}
